unit_tests/L1_testing/test_files/test_Bundleprocessor_ut.py

```diff
@@ -1402,5 +1400,1381 @@ def test_optional_feild_app_meta_data_schema(self):
+
+    def test_get_real_uid_gid(self):
+        logger.debug("-->checking get uid gid")
+        processor = BundleProcessor()
+        processor.rootfs_path = None
+        processor.createmountpoints = None
+        processor.platform_cfg = {
+            "disableUserNamespacing":False,
+            "usersAndGroups":{
+                "uidMap":"",
+                "gidMap":""
+            }
+        }
+        processor.oci_config={
+            "process":{
+                "user":{
+                    "gid": '2',
+                    "uid": '1'
+                }
+            }
+        }
+        processor.get_real_uid_gid()
+        expected = {
+            "process": {
+                "user": {
+                    "gid": '2',
+                    "uid": '1'
+                }
+            }
+        }
+        self.assertEqual(processor.oci_config, expected)
+
+    def test_logging_missing_in_platform(self):
+        logger.debug("-->Platform does not contain logging options")
+        processor = BundleProcessor()
+        processor.rootfs_path = None
+        processor.createmountpoints = None
+        processor.platform_cfg = {
+        }
+        actual = processor._process_logging()
+        self.assertEqual(None, actual)
+
+    def test_checking_storage_in_persistent(self):
+    #if storage_settings are not present
+        logger.debug("-->It will create mounts")
+        processor = BundleProcessor()
+        processor.rootfs_path = "/tmp/test"
+        processor.createmountpoints = None
+        processor.app_metadata = {
+            "id":"com.rdk.flutter",
+            "storage": {
+                "persistent":[{
+                    "size": "10M",
+                    "path": "/home/private"
+                }]
+            }
+        }
+        processor.platform_cfg = {
+            "storage":{
+                "persistent":{
+                    "storageDir": "/opt/dac_apps/data/0/dac",
+                    "maxSize": "20M",
+                    "minSize":"15M",
+                    "maxTotalSize":"10"
+                }
+            }
+        }
+        processor.oci_config = {
+            "mounts":[],
+            "rdkPlugins":{
+                "storage":{
+                    "data":{
+                        "loopback":[]
+                    }
+                }
+            }
+        }
+        processor._process_storage()
+        expected = {
+            "mounts":[],
+            'rdkPlugins': {
+                'storage': {
+                    'data': {
+                        'loopback': [{
+                            'destination': '/home/private',
+                                'flags': 14,
+                                'fstype': 'ext4',
+                                'imgsize': 10485760,
+                                'source': '/opt/dac_apps/data/0/dac/com.rdk.flutter/23e959658f5082a89c6db72c842271a117887b8658ab703f60eaba650b3d5f20.img'}]
```

Comment:
where does the `23e959658f5082a89c6db72c842271a117887b8658ab703f60eaba650b3d5f20.img` name come from ? is it hard coded ?

Reply:
No, it's not hard coded. .img in the source path is formed from the hash_key of dest_path.
https://github.com/rdkcentral/BundleGen/blob/0d282784b9f3d81e766416a8d6ccd20ebf8154fc/bundlegen/core/bundle_processor.py#L861